server/api/region.go

```diff
@@ -662,66 +662,57 @@ func (h *regionsHandler) ScatterRegions(w http.ResponseWriter, r *http.Request)
 	if err := apiutil.ReadJSONRespondError(h.rd, w, r.Body, &input); err != nil {
 		return
 	}
-	var regionMap map[uint64]*core.RegionInfo
 	_, ok1 := input["start_key"].(string)
 	_, ok2 := input["end_key"].(string)
-	regionsCount := 0
+	group, ok := input["group"].(string)
+	if !ok {
+		group = ""
+	}
+	retryLimit, ok := input["retry_limit"].(int)
+	if !ok {
+		retryLimit = 5
+	}
+	var ops []*operator.Operator
+	var failures map[uint64]error
+	var err error
 	if ok1 && ok2 {
 		startKey, _, err := parseKey("start_key", input)
 		if err != nil {
 			h.rd.JSON(w, http.StatusBadRequest, err.Error())
 			return
 		}
-
 		endKey, _, err := parseKey("end_key", input)
 		if err != nil {
 			h.rd.JSON(w, http.StatusBadRequest, err.Error())
 			return
 		}
-		regions := rc.ScanRegions(startKey, endKey, -1)
-		regionMap = make(map[uint64]*core.RegionInfo, len(regions))
-		for _, region := range regions {
-			regionMap[region.GetID()] = region
+		ops, failures, err = rc.GetRegionScatter().ScatterRegionsByRange(startKey, endKey, group, retryLimit)
+		if err != nil {
+			h.rd.JSON(w, http.StatusInternalServerError, err.Error())
+			return
 		}
-		regionsCount = len(regionMap)
 	} else {
 		regionsID := input["regions_id"].([]uint64)
-		regionMap = make(map[uint64]*core.RegionInfo, len(regionsID))
-		for _, id := range regionsID {
-			regionMap[id] = rc.GetRegion(id)
+		ops, failures, err = rc.GetRegionScatter().ScatterRegionsByID(regionsID, group, retryLimit)
```

Comment:
Is this result calculated after the operator finished?

Reply:
The result (like finished percentage) would be calculated in the handler layer.